pkg/router/template/router.go

```diff
@@ -203,6 +203,19 @@ func matchPattern(pattern, s string) bool {
 	return false
 }
 
+// Generate a regular expression to match wildcard hosts (and paths if any)
+// for a [sub]domain.
+func genDomainWildcardRegexp(hostname, path string, exactPath bool) string {
+	route := &routeapi.Route{Spec: routeapi.RouteSpec{Host: hostname}}
+	host, _ := routeapi.NormalizeWildcardHost(route.Spec.Host)
```

Comment:
It feels a little odd to generate wildcards even if the host doesn't have a leading wildcard.  But I don't know how to handle the error path, and we are checking in the config file that they do... so it should be safe.